subgraph/scripts/update.sh

```
@@ -1,9 +1,26 @@
 
-function update() #hardhatNetwork #graphNetwork #subgraphConfig #dataSourceIndex #contract
-{
+function isContractDeployed() { #hardhatNetwork #graphNetwork #subgraphConfig #dataSourceIndex #contract
+    local hardhatNetwork="$1"
+    local graphNetwork="$2"
+    local subgraphConfig="$3"
+    local dataSourceIndex="$4"
+    local dataSourceName="$5"
+
+    # Get the deployment artifact
+    local contractName=$(basename $(yq '.dataSources['$dataSourceIndex'].mapping.abis[] | select(.name == "'$dataSourceName'") | .file' "$subgraphConfig") .json)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)